src/ImageSharp/Formats/Tiff/TiffFrameMetadataResolutionExtensions.cs

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) Six Labors.
+// Licensed under the Apache License, Version 2.0.
+
+using SixLabors.ImageSharp.Common.Helpers;
+using SixLabors.ImageSharp.Metadata;
+using SixLabors.ImageSharp.Metadata.Profiles.Exif;
+
+namespace SixLabors.ImageSharp.Formats.Experimental.Tiff
+{
+    internal static class TiffFrameMetadataResolutionExtensions
+    {
+        public static void SetResolutions(this TiffFrameMetadata meta, PixelResolutionUnit unit, double horizontal, double vertical)
+        {
+            switch (unit)
+            {
+                case PixelResolutionUnit.AspectRatio:
+                case PixelResolutionUnit.PixelsPerInch:
+                case PixelResolutionUnit.PixelsPerCentimeter:
+                    break;
+                case PixelResolutionUnit.PixelsPerMeter:
+                    {
+                        unit = PixelResolutionUnit.PixelsPerCentimeter;
+                        horizontal = UnitConverter.MeterToCm(horizontal);
+                        vertical = UnitConverter.MeterToCm(vertical);
+                    }
+
+                    break;
+                default:
+                    unit = PixelResolutionUnit.PixelsPerInch;
+                    break;
+            }
+
+            meta.SetSingle(ExifTag.ResolutionUnit, (ushort)unit + 1);
+            meta.SetResolution(ExifTag.XResolution, horizontal);
+            meta.SetResolution(ExifTag.YResolution, vertical);
+        }
+
+        public static PixelResolutionUnit GetResolutionUnit(this TiffFrameMetadata meta)
+        {
+            if (!meta.TryGetSingle(ExifTag.ResolutionUnit, out ushort res))
+            {
+                res = TiffFrameMetadata.DefaultResolutionUnit;
+            }
+
+            return (PixelResolutionUnit)(res - 1);
```

Comment:
Use `UnitConverter.ExifProfileToResolutionUnit` here once you implement the meta to use the profile.

Reply:
yes, when implemented with exifProfile, I will make beautiful using `UnitConverter.ExifProfileToResolutionUnit`